contrib/hadoop-store-builder/src/java/voldemort/store/readonly/fetcher/HdfsFetcherAggStats.java

```diff
@@ -0,0 +1,190 @@
+package voldemort.store.readonly.fetcher;
+
+import org.apache.log4j.Logger;
+import voldemort.annotations.jmx.JmxGetter;
+import voldemort.utils.JmxUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+ * This class is used to capture the aggregated metrics for all the file pulls from Hdfs,
+ * and expose them through JMX Bean.
+ */
+public class HdfsFetcherAggStats {
+    // A singleton for all the data pull tasks
+    private static final HdfsFetcherAggStats stats;
+
+    private static Logger logger = Logger.getLogger(HdfsFetcherAggStats.class);
+
+    // Total bytes fetched by all the data pull tasks
+    private long totalBytesFetched;
+    // Total Hdfs fetch retry number
+    private long totalFetchRetries;
+    // Total checksum failures
+    private long totalCheckSumFailures;
+    // Total file read failures
+    private long totalFileReadFailures;
+    // Total authentication failures
+    private long totalAuthenticationFailures;
+    // Total file not found failures
+    private long totalFileNotFoundFailures;
+    // Total quota exceed failures
+    private long totalQuotaExceedFailures;
+    // Total unauthorized store failures
+    private long totalUnauthorizedStoreFailures;
+
+    // Parallel active fetch number
+    // This metric is only used to capture the active fetches number right now
+    private int parallelFetches;
```

Comment:
Rename this to activeFetches to be more general.

parallelFetches can be used later on if we want to control parallelism/priority of fetches and want to track how many are pulling data in parallel.